projects/igniteui-angular/src/lib/grids/tree-grid/tree-grid-indentation.spec.ts

```diff
@@ -0,0 +1,360 @@
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(5), rows[5], 2);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(6), rows[6], 2);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(7), rows[7], 0);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(8), rows[8], 0);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(9), rows[9], 1);
+        });
+
+        it('should persist the indentation after sorting', () => {
+            treeGrid.sort({ fieldName: 'Age', dir: SortingDirection.Asc });
+            fix.detectChanges();
+
+            const rows = TreeGridFunctions.sortElementsVertically(TreeGridFunctions.getAllRows(fix));
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(0), rows[0], 0);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(1), rows[1], 1);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(2), rows[2], 0);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(3), rows[3], 1);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(4), rows[4], 1);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(5), rows[5], 1);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(6), rows[6], 2);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(7), rows[7], 2);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(8), rows[8], 2);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(9), rows[9], 0);
+        });
+
+        it('should persist the indentation after filtering', () => {
+            treeGrid.filter('Age', 40, IgxNumberFilteringOperand.instance().condition('greaterThan'));
+            fix.detectChanges();
+
+            const rows = TreeGridFunctions.sortElementsVertically(TreeGridFunctions.getAllRows(fix));
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(0), rows[0], 0);
+
+            // This row does not satisfy the filtering, but is present in the DOM with lowered opacity
+            // in order to indicate that it is a parent of another record that satisfies the filtering.
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(1), rows[1], 1);
+
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(2), rows[2], 2);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(3), rows[3], 0);
+            TreeGridFunctions.verifyRowIndentationLevel(treeGrid.getRowByIndex(4), rows[4], 0);
+        });
+
+        it('should persist the indentation on all pages when using paging', () => {
+            treeGrid.paging = true;
```

Comment:
Both properties can be changed before detecting the changes.

Reply:
Fixed